mne_bids/tests/test_write.py

```diff
@@ -4187,3 +4187,40 @@ def test_write_evt_metadata(_bids_validate, tmp_path):
     for cur_col in event_metadata.columns:
         assert cur_col in events_tsv
         assert cur_col in events_json
+
+
+@testing.requires_testing_data
+def test_write_bids_with_age_weight_info(tmp_path):
+    """Test writing participant.tsv when using np.arrays for weight and height."""
+    bids_root = tmp_path / "bids"
+    raw_fname = data_path / "MEG" / "sample" / "sample_audvis_trunc_raw.fif"
+    raw = _read_raw_fif(raw_fname)
+    raw.info["subject_info"] = {
+        "weight": np.array([75.0]),
+        "height": np.array([180.0]),
+    }
```

Comment:
I actually just thought of an issue... whenever https://github.com/mne-tools/mne-python/issues/12874 is tackled (which I was about to try to see if it's trivial!) this will raise a `ValueError`. So I think you need to bypass the soon-to-be-added MNE protection:
```suggestion
    # bypass MNE-Python setter protection
    dict.__setitem__(
        raw.info,
        "subject_info",
        {
            "weight": np.array([75.0]),
            "height": np.array([180.0]),
        },
    )
```

Reply:
I'm afraid this does not do the trick because while the initial setting works, the `write_raw_bids` function copies the `raw` object  [here](https://github.com/mne-tools/mne-bids/blob/f94c24ee241ee214b65f776e7197f031e206e36c/mne_bids/write.py#L1737). And this calls the validation again. I have not found any way around it, including trying to mock the `__setitems__` method to be the original one from `dict`.

Reply:
Pushed a commit using `monkeypatch` that should work

Reply:
so that monkeypatch will essentially always have to be there now?

Reply:
For the test to make sure that MNE-BIDS has its own validation, yes